projects/plugins/boost/app/modules/Modules_Setup.php

```diff
@@ -168,6 +169,31 @@ public function setup() {
 		$this->setup_modules_data_sync( $this->modules_index->get_modules() );
 		add_action( 'plugins_loaded', array( $this, 'load_modules' ) );
 		add_action( 'jetpack_boost_module_status_updated', array( $this, 'on_module_status_update' ), 10, 2 );
+
+		// Add a hook to fire page output changed action when a module that Changes_Output_After_Activation indicate something has changed.
+		foreach ( $this->available_modules as $module ) {
+			$feature = $module->feature;
+			if ( $module->is_enabled() && $feature instanceof Changes_Output_After_Activation ) {
+				$action_names = $feature::get_change_output_action_names();
+				if ( ! empty( $action_names ) ) {
+					foreach ( $action_names as $action ) {
+						add_action( $action, array( $this, 'handle_module_output_change' ), 10, 1 );
+					}
+				}
+			}
```

Comment:
What do you think about this? I'm not a fan of nesting 😅 

```suggestion
			if ( ! $module->is_enabled() ) {
				continue;
			}

			$feature = $module->feature;
			if ( ! ( $feature instanceof Changes_Output_After_Activation ) ) {
				continue;
			}

			$action_names = $feature::get_change_output_action_names();
			if ( empty( $action_names ) ) {
				continue;
			}

			foreach ( $action_names as $action ) {
				add_action( $action, array( $this, 'handle_module_output_change' ), 10, 1 );
			}
```

Reply:
Makes sense, I'll update accordingly.